### ISSUE TITLE
[Forge 2708]NullPointerException

### ISSUE THREAD

**Issue by Pythonic-Rainbow** [TheCBProject/NotEnoughItems#162]
Crash report: https://pastebin.com/mrzA6RVp
Full log: https://1drv.ms/t/s!AuxrrFvXeZw1g4gcQSvK3fk8XUpeRg

**Comment by covers1624** [comment]
Going to need a full log, there is literally no stack trace in that crash report.

**Comment by Pythonic-Rainbow** [comment]
@covers1624 sure

**Comment by Pythonic-Rainbow** [comment]
In case anyone needs to know, the problem isn't on Thamuic JEI. I removed it and still crashes

**Comment by Pythonic-Rainbow** [comment]
Tested in 2708. Same result.

**Comment by grydian** [comment]
Getting the same crash.
https://pastebin.com/XJgHuXUe

**Comment by Ligands** [comment]
@covers1624, this is related to / another incident of the underlying problem that caused #161. It's a multi-mod incompatibility issue relating to Astral Sorcery + many other mods- however, it doesn't occur with simply NEI & AS installed. This particular incident is caused by IC2, I've just replicated it myself, here is the relevant stack trace:

```
net.minecraftforge.fml.common.LoaderExceptionModCrash: Caught exception from Not Enough Items (nei)
Caused by: java.lang.NullPointerException
    at ic2.core.init.BlocksItems$1.getItemStackLimit(BlocksItems.java:430)
    at net.minecraft.item.ItemStack.getMaxStackSize(ItemStack.java:248)
    at hellfirepvp.astralsorcery.common.enchantment.amulet.EnchantmentUpgradeHelper.isItemBlacklisted(EnchantmentUpgradeHelper.java:176)
    at hellfirepvp.astralsorcery.common.enchantment.amulet.PlayerAmuletHandler.attachAmuletItemCapability(PlayerAmuletHandler.java:38)
    at net.minecraftforge.fml.common.eventhandler.ASMEventHandler_153_PlayerAmuletHandler_attachAmuletItemCapability_AttachCapabilitiesEvent.invoke(.dynamic)
    at net.minecraftforge.fml.common.eventhandler.ASMEventHandler.invoke(ASMEventHandler.java:90)
    at net.minecraftforge.fml.common.eventhandler.EventBus.post(EventBus.java:182)
    at net.minecraftforge.event.ForgeEventFactory.gatherCapabilities(ForgeEventFactory.java:658)
    at net.minecraftforge.event.ForgeEventFactory.gatherCapabilities(ForgeEventFactory.java:640)
    at net.minecraft.item.ItemStack.forgeInit(ItemStack.java:1216)
    at net.minecraft.item.ItemStack.<init>(ItemStack.java:112)
    at net.minecraft.item.ItemStack.<init>(ItemStack.java:98)
    at codechicken.lib.inventory.InventoryUtils.newItemStack(InventoryUtils.java:60)
    at codechicken.nei.util.ItemStackMap.wildcard(ItemStackMap.java:224)
    at codechicken.nei.util.ItemStackSet.with(ItemStackSet.java:23)
    at codechicken.nei.init.NEIInitialization.loadModSubsets(NEIInitialization.java:329)
    at codechicken.nei.init.NEIInitialization.bootNEI(NEIInitialization.java:72)
```

Instances of this bug been springing up all over the place since **Astral Sorcery's 1.9.0** update in various shapes and forms, not just NPEs- they added some code that's checking all items for enchantments or something:

- https://github.com/HellFirePvP/AstralSorcery/issues/754
- https://github.com/HellFirePvP/AstralSorcery/issues/749
- https://github.com/HellFirePvP/AstralSorcery/issues/747

I first experienced the bug with **Mekanism** myself & the crash logs seemed to point to NEI being the culprit, but I discovered that removing NEI didn't fix the issue. Mekanism has released a fix for their issue- [Mekanism 9.4.12.348](https://minecraft.curseforge.com/projects/mekanism/files/2572553).

However- I'm still experiencing the issue myself with both **Unique Crops** (v1.12-0.1.52) & **IC2** (v2.8.84-ex112), and have heard that it also affects the **Erebus** at least. In these instances, removing NEI _does_ fix the crash- so NEI may still be a suspect.

Correct me if I'm wrong, but from what I can tell, it's likely relating to this part, as (it sounds like)[https://github.com/aidancbrady/Mekanism/issues/5081#issuecomment-397187243] some mods might be unprepared to provide wildcard items for whatever AS is doing:
```
...
    at codechicken.nei.util.ItemStackMap.wildcard(ItemStackMap.java:224)
    at codechicken.nei.util.ItemStackSet.with(ItemStackSet.java:23)
    at codechicken.nei.init.NEIInitialization.loadModSubsets(NEIInitialization.java:329)
...
```

@Pythonic-Rainbow can we please change the title to 'Astral Sorcery 1.9.0 incompatibility' or something a bit more descriptive?

**Comment by covers1624** [comment]
In that case, duplicate of https://github.com/TheCBProject/NotEnoughItems/issues/161  
There is nothing i can do, i create an itemstack and their event handler explodes.

**Comment by Ligands** [comment]
@covers1624 fair enough, just trying to help, as removing NEI fixes the issue and all.

I'll open a ticket with Unique Crops now. @Pythonic-Rainbow you'll need to open an issue with IC2 for your problem.

**Comment by Pythonic-Rainbow** [comment]
@Ligands sure

**Comment by Pythonic-Rainbow** [comment]
IC2 issue: https://bt.industrial-craft.net/view.php?id=2383

**Comment by Pythonic-Rainbow** [comment]
IC2 says the problem is resolved in 2.2.85 @covers1624
